common/trieForDirPath.go

```diff
@@ -0,0 +1,83 @@
+package common
+
+import (
+	"strings"
+)
+
+type TrieNode struct {
+	Children map[string]*TrieNode
+	Value    uint32
+	isEnd    bool
+}
+
+type Trie struct {
+	Root *TrieNode
+}
+
+func NewTrie() *Trie {
+	return &Trie{
+		Root: &TrieNode{Children: make(map[string]*TrieNode)},
+	}
+}
+
+func (t *Trie) Insert(key string, value uint32) {
+	node := t.Root
+	segments := strings.Split(key, "/")
+	for _, segment := range segments {
+		child, exists := node.Children[segment]
+		if !exists {
+			child = &TrieNode{Children: make(map[string]*TrieNode)}
+			node.Children[segment] = child
+		}
+		node = child
+	}
+	node.Value = value
+	node.isEnd = true
+}
+
+func (t *Trie) Get(key string) (uint32, bool) {
+	node := t.Root
+	segments := strings.Split(key, "/")
+	for _, segment := range segments {
+		child, exists := node.Children[segment]
+		if !exists {
+			return 0, false
+		}
+		node = child
+	}
+	if node.isEnd {
+		return node.Value, true
+	}
+	return 0, false
+}
+
+func (t *Trie) Delete(key string) bool {
```

Comment:
this can be addressed after the PR is merged but Delete is no longer used right? So it can be deleted

Reply:
we are using this delete function for removing the entries from unregisteredButCreated